File: frontend/src/microservices/operations/objects/ManagerRefundTaskResponse.ts
// 本文件定义 operations 模块的 `ManagerRefundTaskResponse`，作为响应数据并提供 JSON 编解码。

export type ManagerRefundTaskResponse = {
  orderId: string
  buyerUserId: string
  taskType: string
  summaryLabel: string
  refundId: string
  refundReason: string
  refundAmount: string
  refundCurrency: string
  requestedAt: string
}
export const managerRefundTaskResponseFromJson = (json: string): ManagerRefundTaskResponse =>
  JSON.parse(json) as ManagerRefundTaskResponse

export const managerRefundTaskResponseToJson = (value: ManagerRefundTaskResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/overall/objects/ManagerRefundTaskResponse.scala
// 本文件定义 `ListManagerRefundTasksPlanner` 返回的单条退款任务数据。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerRefundTaskResponse(
    orderId: String,
    buyerUserId: String,
    taskType: String,
    summaryLabel: String,
    refundId: String,
    refundReason: String,
    refundAmount: String,
    refundCurrency: String,
    requestedAt: String
)
object ManagerRefundTaskResponse:
  given sourceEncoder: Encoder[ManagerRefundTaskResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerRefundTaskResponse] = deriveDecoder
